x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.test.tsx

```diff
@@ -248,11 +263,184 @@ describe('IndexPattern Data Source', () => {
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging: { name: 'bar' },
+              }}
+              state={dragDropState()}
+              setState={() => {}}
+              columnId={'col2'}
+              filterOperations={() => true}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          expect((dropTarget as { droppable: boolean }).droppable).toBeFalsy();
+        });
+
+        it('is not droppable if field is not supported by filterOperations', () => {
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging: { type: 'number', name: 'bar' },
+              }}
+              state={dragDropState()}
+              setState={() => {}}
+              columnId={'col2'}
+              filterOperations={() => false}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          expect((dropTarget as { droppable: boolean }).droppable).toBeFalsy();
+        });
+
+        it('is droppable if the field is supported by filterOperations', () => {
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging: { type: 'number', name: 'bar' },
+              }}
+              state={dragDropState()}
+              setState={() => {}}
+              columnId={'col2'}
+              filterOperations={op => op.dataType === 'number'}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          expect((dropTarget as { droppable: boolean }).droppable).toBeTruthy();
+        });
+
+        it('appends the dropped column when a field is dropped', () => {
+          const dragging = { type: 'number', name: 'bar' };
+          const testState = dragDropState();
+          const setState = jest.fn();
+          const component = shallow(
+            <IndexPatternDimensionPanel
+              dragDropContext={{
+                ...dragDropContext,
+                dragging,
+              }}
+              state={testState}
+              setState={setState}
+              columnId={'col2'}
+              filterOperations={op => op.dataType === 'number'}
+            />
+          );
+
+          const dropTarget = component
+            .find('[data-test-subj="indexPattern-dropTarget"]')
+            .first()
+            .props();
+
+          const onDrop = (dropTarget as { onDrop: DropHandler }).onDrop;
+
+          onDrop(dragging);
```

Comment:
I think you can really simplify your code here. Typescript already knows the function type, but you need to tell it that `onDrop` is defined: `component.find('...').first().prop('onDrop')!(dragging);`